app/dashboard/helpers.py

```diff
@@ -467,6 +467,29 @@ def process_bounty_details(bounty_details):
     return (did_change, latest_old_bounty, latest_old_bounty)
 
 
+def record_user_action(event_name, old_bounty, new_bounty):
+    user_profile = None
+    try:
+        if event_name in ['new_bounty', 'killed_bounty', 'work_done']:
+            user_profile = Profile.objects.get(handle=new_bounty.bounty_owner_github_username)
+        if event_name in ['work_submitted']:
+            fulfillment = bounty.fulfillments.order_by('pk').first()
+            user_profile = Profile.objects.get(handle=fulfillment.fulfiller_github_username)
+    except Exception as e:
+        logging.error(f'{e} during record_user_action for {new_bounty}')
+        # TODO: create a profile if one does not exist already?
```

Comment:
We could handle this with a `get_or_create` with something like:

```python
def record_user_action(event_name, old_bounty, new_bounty):
    user_profile = None

    try:
        if event_name in ['new_bounty', 'killed_bounty', 'work_done']:
            handle = new_bounty.bounty_owner_github_username
        elif event_name in ['work_submitted']:
            fulfillment = new_bounty.fulfillments.order_by('pk').first()
            handle = fulfillment.fulfiller_github_username

        user_profile, _ = Profile.objects.get_or_create(
            handle=handle,
            defaults={'data': {}}
        )
    except Exception as e:
        logging.error(f'{e} during record_user_action for {new_bounty}')

    if user_profile:
        UserAction.objects.create(
            profile=user_profile,
            action=event_name,
            metadata={
                'new_bounty': new_bounty.pk if new_bounty else None,
                'old_bounty': old_bounty.pk if old_bounty else None,
            })
```

This reminds me...  We could probably update `sync_profile` to use `Profile.objects.update_or_create()` and return the profile.

Reply:
> We could handle this with a get_or_create with something like:

+1 on this idea.. its a very good one.  i almost feel like we could do one-better and call `sync_profile()` after the get_or_create so that way we have the `data` `repo_data` etc fields available.  

here is a commit that does so.. what do you think? 60799407a849a5ea3ea697e511f85f08eb33172d

> This reminds me... We could probably update sync_profile to use Profile.objects.update_or_create() and return the profile.

doesnt the get_or_create in that method prettymuch accomplish this already though?  or you mean that it should update the `data` `repo_data` etc fields?  seems like this lines alraedy does this.. although maybe less performantly:

```
    org.handle = handle
    org.data = data
    org.repos_data = repos_data
    org.save()
```